### ISSUE TITLE
support for dark mode

### ISSUE THREAD

**Issue by TruncatedDinoSour** [mailvelope/mailvelope#834]
hello,

it'd be nice if mailvelope supported dark mode, i tried to add it myself, though as i'm nowhere close to a js dev i kept getting confused, closest i got was adding a dark mode toggle and then i kept getting lost, so would be cool if either dark mode was added, or i could get some help on https://github.com/TruncatedDinoSour/mailvelope/tree/dev :)

thanks

**Comment by TruncatedDinoSour** [comment]
https://github.com/TruncatedDinoSour/mailvelope/commit/908fcbb07e34160f79d0b026588899a68c38bfc7#diff-21f150f778a241662df24476e505382a8cadb996dffd5fcd7d529e165e7a8cefR113

my idea was to add a 'dark' class for dark mode and change css variables accordingly, but this doesn't work

**Comment by rimmer** [comment]
Hey @TruncatedDinoSour, thanks for your efforts! I would say, the most complex part is to actually implement dark mode everywhere and make sure it does not cause issues. Specifically, important stuff is visible and the security background feature does not suffer. 
AFAIK, we do not plan to implement dark more in the near future, mainly because we are a very small team and trying to focus on what is most important.

**Comment by rimmer** [comment]
Feel free to close the issue if you are not planning to work on it, otherwise we would be happy to see your PR :)

**Comment by TruncatedDinoSour** [comment]
> Hey @TruncatedDinoSour, thanks for your efforts! I would say, the most complex part is to actually implement dark mode everywhere and make sure it does not cause issues. Specifically, important stuff is visible and the security background feature does not suffer. AFAIK, we do not plan to implement dark more in the near future, mainly because we are a very small team and trying to focus on what is most important.

hi :D thanks for the reply

it is sad to hear that you don't plan to implement dark mode, but for the time being, maybe i'll keep the issue open, maybe someone else will find it of interest or maybe in the far future opinions change, well see

**Comment by TruncatedDinoSour** [comment]
> Feel free to close the issue if you are not planning to work on it, otherwise we would be happy to see your PR :)

oh !

alright then, ill try to figure it out again, hopefully, last time it didnt go so well, but ill try my best :D

**Comment by TruncatedDinoSour** [comment]
me and @t1nklas ( and maybe @TheCubiq if she has time, i invited her ) are working on the new fork trying to figure it out lmao, https://github.com/TruncatedDinoSour/mailvelope :)

**Comment by rimmer** [comment]
While going through issues today, I noticed another request for Dark Mode #799.

**Comment by rimmer** [comment]
Just a reminder, that modern browsers can report selected mode with [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) CSS property.

**Comment by TruncatedDinoSour** [comment]
> While going through issues today, I noticed another request for Dark Mode #799.

yeah many people would benefit from dark mode i believe, i am sensitive to light as well, though not highly sensitive, ive currently come up with this theme :

```scss
$white: #1a1a1a;
$gray-100: #2a2a2a; // secondary-button-bg
$gray-200: #3a3a3a; // columnheader-bg
$gray-300: #4a4a4a; // table-pressed
$gray-400: #5a5a5a; // table-divider
$gray-500: #6a6a6a; // grey-hover
$gray-600: #7a7a7a; // grey86
$gray-700: #8a8a8a; // prompttext
$gray-800: #9a9a9a; // grey46
$gray-900: #b0b0b0; // lighter gray for text
$black: #e0e0e0; // Light text color

$green: #4dd1a8;
$green-dark: #5a9a7a; // footer-text
$red: #ff4c8f;
$orange: #ffb84d;
$cyan: #b3d1d3;
$dark-cyan: #a3c6c7;
$turquoise: #a3b3b3;
```

i believe the contrast is good as well as being dark

**Comment by rimmer** [comment]
> everywhere theres only scss variables which get substituted at build time which is so annoying, so im considering turning pretty much everything into a property but i dont know how much of a good idea this is

did you actually manage to run the extension locally? I mean, if compiled and tested locally, scss is not much of an issue

**Comment by TruncatedDinoSour** [comment]
> > everywhere theres only scss variables which get substituted at build time which is so annoying, so im considering turning pretty much everything into a property but i dont know how much of a good idea this is
> 
> did you actually manage to run the extension locally? I mean, if compiled and tested locally, scss is not much of an issue or maybe I did not understand the issue you are facing

i have, `npm ci`, then grunt, and then grunt for firefox ( https://github.com/mailvelope/mailvelope?tab=readme-ov-file#build-instructions )

the issue im seeing is it inlining scss variables instead of them being properties, for example we have

```scss
$white: #fff;
```

and it is used in

```scss
.file-panel {
  .file-element {
    display: inline-flex;
    flex-direction: column;
    padding: 0.45rem 0.5rem;
    margin-bottom: 0.5rem;
    background-color: $white;
...
```

after packaging it, we got this :

![image](https://github.com/user-attachments/assets/c7df80c1-3e93-4799-a83f-743134689094)

which is not the same as using `var(--white)` which would be very helpful in doing dark mode, as now we are either stuck at making everything a CSS property ( using `var()` ) or rewriting all parts of scss which uses colours like this to overwrite it using !important or something i think

**Comment by TruncatedDinoSour** [comment]
@rimmer do you have opinions on turning everything into a custom css property when it comes to colours ? i believe this is the only feasable way to implement it in a way which makes sense

**Comment by rimmer** [comment]
> @rimmer do you have opinions on turning everything into a custom css property when it comes to colours ? i believe this is the only feasable way to implement it in a way which makes sense

Hi @TruncatedDinoSour 
Thank you for the explanation!
We discussed it internally and it looks good👍🏻 to use css variables for theming. Regarding old browser versions -- my personal take is that if someone uses outdated browser version they don't care about dark theme or they know how to handle it themselves, so I don't think we need to add a polyfill.

**Comment by rimmer** [comment]
> tbf i havent truly touched scss in some time now but i kinda have a faint memory of being able to turn scss variables into a css properties easily somehow or having runtime values in some way or another, but i dont remember how exactly - will need to dig into it

I kind of googled it recently and looks like the easiest way is in fact to use css variables (but I didn't dig too deep). I think there was a mention of a plugin that does the conversion, but it will just add complexity

**Comment by TruncatedDinoSour** [comment]
theres a big problem now : my keyboard is very broken right now and refuses to type unless i jam every key down as hard as i can

this happened in the past and it just fixed itself but im hoping either it fixes itself today / soon so i could wok on his, or ill e forrrrrrced to waitt til septeme

vvery sorry fforrr this slow progess

**Comment by rimmer** [comment]
> theres a big problem now : my keyboard is very broken right now and refuses to type unless i jam every key down as hard as i can
> 
> this happened in the past and it just fixed itself but im hoping either it fixes itself today / soon so i could wok on his, or ill e forrrrrrced to waitt til septeme
> 
> vvery sorry fforrr this slow progess

Don't worry, take your time!

**Comment by TruncatedDinoSour** [comment]
@rimmer i got a new keyboard today, ill start work on this tmrw :)

**Comment by rimmer** [comment]
> @rimmer i got a new keyboard today, ill start work on this tmrw :)

Congrats with the new keyboard, I hope it is better than mine 😁 ! Thank you for taking the initiative!

**Comment by TruncatedDinoSour** [comment]
> @rimmer i got a new keyboard today, ill start work on this tmrw :)

i keep promising that ill work on this but keep getting busy with school stuff, and my priorities tell me that its more important than mailvelope xD

ill see if i can shove it into my schedule in the next 3 days sometime, i mean, weekends are less busy so i might be able to do it friday evening or saturday morning, today school stuff still ugh i am truly sorry for all the wait i am causing, i really want to get to it but life is very busy for me at the moment with a load of homework and independent work

mailvelope is on my highest priority to do after school, but i cant get to it easily for now, but i promise that it is coming soon <3

**Comment by TruncatedDinoSour** [comment]
ive been working on this today and ive realise that mailvelope scss sets some variables used by bootstrap which also uses scss functions so it needs compile-time colour codes, so, how shall we go about this ?

maybe we just overwrite stuff using `!important` ? i mean, thats such a terribly dirty solution ...

data-bs-theme=dark also isnt helping as far as i can see, which sucks, i am hella lost, why is front-end stuff so complicated :sob:
